riscv-runtime/src/goldilocks.rs

```diff
@@ -1,13 +1,61 @@
-use core::arch::asm;
+use core::{arch::asm, mem::MaybeUninit};
 use powdr_riscv_syscalls::Syscall;
 
 pub const PRIME: u64 = 0xffffffff00000001;
 
+/// Goldilocks field element.
+///
+/// Not a legal value in RISC-V, as it uses a supposedly
+/// 32-bit memory word to store a full field element.
```

Comment:
Shouldn't we use usize then?

Reply:
Probably not. The rust type can be anything that has the size and alignment of our implementation of one memory word. `usize` is identical, in those regards, to `u32`.

But suppose we are implementing RISCV-64: would we still use our 32-bit memory word implementation? Probably yes, in the Goldilocks field. So the correct type would still be `u32`, but not `usize`.

So, `u32` seems better.

Reply:
Ok.
But regarding RV64, why would we implement it as 32-bit memory words?

Reply:
It is the largest power of two that fits Goldilocks. These memory machine words are more a restriction of the field than the implemented VM, right?

Reply:
Well if we implemented RV64 with GL we'd use two limbs, like we do with BB for RV32

Reply:
Why BB was done like that instead of using words of 16 bits? To avoid too many changes in the RV32 machine?

Reply:
Not sure I get the question, the BB RISCV memory machine was written exactly like that: 2 FE limbs of 16 bits